src/Client.js

```diff
@@ -75,7 +75,11 @@ class Client extends EventEmitter {
                      */
                     this.emit(Events.AUTHENTICATION_FAILURE, 'Unable to log in. Are the session details valid?');
                     browser.close();
-
+                    if (this.options.restartOnAuthFail) {
```

Comment:
Should be: this.options.puppeteer.restartOnAuthFail

Reply:
Why? We're checking the options passed to client()...

Reply:
This seems fine to me